packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2700/onlp/builds/src/module/src/sysi.c

```diff
@@ -0,0 +1,288 @@
+        DEBUG_PRINT("[Debug][%s][%d]Failed to read output of command '%s'\n",
+                    __FUNCTION__, __LINE__, command);
+        pclose(fp);
+    }
+
+    /* The last symbol is '\n', so remote it */
+    buffer[strnlen(buffer, COMMAND_OUTPUT_BUFFER) - 1] = '\0';
+
+    /* close */
+    pclose(fp);
+}
+
+const char*
+onlp_sysi_platform_get(void)
+{
+    return "x86-64-mlnx-msn2700-r0";
+}
+
+int
+onlp_sysi_onie_data_get(uint8_t** data, int* size)
+{
+    uint8_t* rdata = aim_zmalloc(256);
+    if(onlp_file_read(rdata, 256, size, IDPROM_PATH) == ONLP_STATUS_OK) {
+        if(*size == 256) {
+            *data = rdata;
+            return ONLP_STATUS_OK;
+        }
+    }
+
+    aim_free(rdata);
+    *size = 0;
+    return ONLP_STATUS_E_INTERNAL;
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{
+    int   i, v[NUM_OF_CPLD]={0};
+
+    for (i=0; i < NUM_OF_CPLD; i++) {
+        v[i] = 0;
+        if(onlp_file_read_int(v+i, "%s/%s", PREFIX_PATH_ON_CPLD_DEV, arr_cplddev_name[i]) < 0) {
+            return ONLP_STATUS_E_INTERNAL;
+        }
+    }
+    pi->cpld_versions = aim_fstrdup("brd=%d, mgmt=%d, port=%d", v[0], v[1], v[2]);
+
+    return ONLP_STATUS_OK;
+}
+
+void
+onlp_sysi_platform_info_free(onlp_platform_info_t* pi)
+{
+    aim_free(pi->cpld_versions);
+}
+
+
+int
+onlp_sysi_oids_get(onlp_oid_t* table, int max)
+{
+    int i;
+    onlp_oid_t* e = table;
+    memset(table, 0, max*sizeof(onlp_oid_t));
+
+    /* 8 Thermal sensors on the chassis */
+    for (i = 1; i <= NUM_OF_THERMAL_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* 6 LEDs on the chassis */
+    for (i = 1; i <= NUM_OF_LED_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 2 PSUs on the chassis */
+    for (i = 1; i <= NUM_OF_PSU_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    /* 8 Fans and 2 PSU fans on the chassis */
+    for (i = 1; i <= NUM_OF_FAN_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+static int
+_onlp_sysi_grep_output(char value[256], const char *attr, const char *tmp_file)
+{
+    int value_offset  = 30; /* value offset in onie-syseeprom */
+    char command[256] = {0};
+    char buffer[COMMAND_OUTPUT_BUFFER]  = {0};
+    int v = 0;
+
+    snprintf(command, sizeof(command), "cat '%s' | grep '%s'", tmp_file, attr);
+    _onlp_sysi_execute_command(command, buffer);
+
+    /* Reading value from buffer with command output */
+    while (buffer[value_offset] != '\n' &&
+           buffer[value_offset] != '\r' &&
+           buffer[value_offset] != '\0') {
+        value[v] = buffer[value_offset];
+        v++;
+        value_offset++;
+    }
+    value[v] = '\0';
+
+    DEBUG_PRINT("[Debug][%s][%d]Value for sytem attribute '%s' is '%s' \n",
+                __FUNCTION__, __LINE__, attr, value);
+
+    return ONLP_STATUS_OK;
+}
+
+int
+onlp_sysi_onie_info_get(onlp_onie_info_t* onie)
```

Comment:
This is typically accomplished by exposing the ONIE eeprom device directly and letting the upper layers read and parse the TLVs. 

Do you have the eeprom device available at all (I don't see it under bsp)? If you do I would suggest making it available here. 

If its not available, then we're going to have to change this around a bit. This call is made frequent enough at runtime that it can't be implemented with these calls to onie-shell without causing a noticable performance issue. 

If the sys-eeprom contents must be generated by calling onie-syseeprom from the onie-shell then this sequence should be moved to the OnlPlatform::baseconfig() method to generate a JSON verison of the system eeprom contents once at boot time. Then this code would be modified to read the contents of the JSON file instead. 

If we have to go this route I will modify the ONIE routines in the ONLPLIB module to support reading directly from the JSON file to populate this structure (it already provides output to JSON, just not input from JSON) to make this easier for you.